rule/expr.go

```diff
@@ -320,6 +320,11 @@ type Param struct {
 	Name string `json:"name"`
 }
 
+//PushExpr implements the Expr interface, but will panic if called as Param's can't have subexpressions.
+func (v *Param) PushExpr(e Expr) {
```

Comment:
Is there a way to avoid that by playing with the fact that Param and Value don't implement the Operator interface?

Reply:
Absolutely - I should have removed them when I changed the interfaces around.   Good catch.